### DISCUSSION TITLE
Sublime Text extension?

### DISCUSSION THREAD

**Discussion by Gabriel-p** [community/community#6617] [Copilot Conversations] [answered]
It would be **great** if the team developed a Sublime Text extension for those of us who are hooked to ST

**Comment by hsenag** [comment, marked as answer]
Thanks for the feedback! We don't have any immediate plans for this but will certainly consider it for the future.

**Reply by LuBreGit** [reply]
**+1**

As a long time ST3 user I'd really really need it

**Reply by FullStackEngineerHassanIlyas** [reply]
> Thanks for the feedback! We don't have any immediate plans for this but will certainly consider it for the future.

@hsenag we really need this extension for ST, any initial steps would be appreciated.

**Reply by cate02** [reply]
bro please you've been considering for a decade please man please

**Reply by TerminalFi** [reply]
@cate02 any reason the extension we’ve developed won’t work for you? It’s pretty much feature complete.

**Reply by cate02** [reply]
Oh yeah gave a quick try installing it but didn't work, though I haven't yet properly tried and fully read your diagnostic tips. I've given up for tonight and so upset that it couldn't be easier by having ST natively supported I came to complain to the copilot guys
Though I hope with less than an hour I could get it all working and well
Also did try VSC for a tiny bit but soon buggered away from that because it was tricky figuring how to get my java app to compile and run to give a console window as well as show the gui, and VSC also complained that the file didn't have it's project folder or some rubbish --- so I just go back to sublime text since it's damn simple, damn competent, and damn nice

**Reply by timfjord** [reply]
I don't think that the official ST extension will be significantly different, as `LSP-copilot` utilises pretty much everything that Sublime Text API can offer

**Reply by TerminalFi** [reply]
If you’ve installed LSP and the LSP-copilot. It should be pretty seamless. The LSP-copilot team ( @timfjord is a primary contributor as well) are happy to work through whatever you might be having issues with.

**Comment by miumerfarooq** [comment]
Thanks, bro we really need that for Sublime.

**Comment by AndreyAzimov** [comment]
Any updates? We all really need it for Sublime.

**Reply by hsenag** [reply]
None at present, sorry.

**Reply by AndreyAzimov** [reply]
F

**Comment by TerminalFi** [comment]
@hsenag is this something you would work with a ST Plugin Dev to do? I started to play around and I believe we can use Sublimes LSP implementation for the major of it since it already supports node packages. I am happy to work on this

**Reply by hsenag** [reply]
@TheSecEng at present we don't have any capacity for this, but in the longer-term helping the community build plugins is very much on our mind.

**Reply by TerminalFi** [reply]
Interesting, are we allowed to do it using the /dist on our own?

**Reply by hsenag** [reply]
I can't encourage that - amongst other things, all the internal interfaces are undocumented and subject to change while we figure out the best way to do things.

**Reply by jfcherng** [reply]
> I can't encourage that

Sorry to bother but since there is no roadmap for other editors and "it will never happen" is an option too, are we "allowed" (rather than "encouraged" or not) to publicly create/publish our own Copilot plugin for other editors with https://github.com/github/copilot.vim/tree/release/copilot/dist ?

**Comment by MillhioreBT** [comment]
I look forward to the github team helping out with contributing GitHub Copilot compatibility for ST4.
It would be a shame if it never came, I would feel that they hate ST4 :(

**Comment by TerminalFi** [comment]
I’ve got a plug-in and will be publishing shortly

**Reply by timfjord** [reply]
@TheSecEng how is the implementation going?  
I've been also playing with it(mainly with the visual part) so maybe I could help you with the package somehow?

**Reply by TerminalFi** [reply]
Myself and another individual have it pretty much done. A few things not done but it works. I’ll see if we will release today. It leverages sublimes LSP plug-in.

**Reply by timfjord** [reply]
Awesome, thanks for the update!

**Reply by TerminalFi** [reply]
https://github.com/TheSecEng/LSP-copilot

Please note we are still working out some bugs and UI issues

**Reply by TerminalFi** [reply]
Find me in the sublime text discord for easier communication

**Comment by TerminalFi** [comment]
https://github.com/TheSecEng/LSP-copilot

Still is WIP

**Reply by timfjord** [reply]
Cool, thanks for the sneak peek.

Regarding the UI, is it final?
<img width="260" alt="image" src="https://user-images.githubusercontent.com/471335/178271924-91c5f0fd-391e-44af-aeb3-844ce889b930.png">
The biggest problem, is that it is currently hidden behind the completion popup

When I played with the phantoms API I managed to get the following UI (please note, the suggestion is just some hardcoded text and not rendered by LSP)
<img width="269" alt="image" src="https://user-images.githubusercontent.com/471335/178273036-f1181478-6091-4d7d-9c5a-1f8ddacae2eb.png">

**Reply by jfcherng** [reply]
Not sure whether it's feasible but even that, it will be still behind the autocompletion box. Popup can live aside the autocompletion box but that is still a bit weird UI.

**Reply by timfjord** [reply]
Yes, all lines but the first will be behind the box, which is acceptable.  
I haven't used Copilot a lot, but most of the time I used it I got suggestions for the first line only.

But I am not sure the solution I've been playing with will be possible (the example I showed was very synthetic).  
Anyway, I will send a PR if I have something to share

**Comment by TerminalFi** [comment]
Feel free to open a PR or create an issue with the approach you took. We
are still trying to determine best path for UI

**Comment by Ricsimond8774** [comment]
Help me recover please

**Reply by TerminalFi** [reply]
? Help you install?

**Reply by MillhioreBT** [reply]
Can you explain how it is installed, some people don't know much. per example: I

**Reply by TerminalFi** [reply]
https://stackoverflow.com/questions/23026201/sublime-text-3-how-to-install-plugins-from-github

**Comment by khengyun** [comment]
> 

thank for ask

**Comment by jfcherng** [comment]
For Sublime Text >= 4126, a unofficial implementation is now on Package Control: https://packagecontrol.io/packages/LSP-copilot

**Comment by isabelc** [comment]
Copilot native support for Sublime Text is still very much wanted by the community. Please.

**Reply by jfcherng** [reply]
The official can't do it better than the community one due to ST's API limitation. But it would be still good for sure if Microsoft pays employees for writing a ST plugin 🤣

**Reply by TerminalFi** [reply]
@isabelc why does it need official support. What is not working that you need official support ?
